packages/web-components/src/tabs/tabs.ts

```diff
@@ -0,0 +1,186 @@
+  /**
+   * activeTabOffset
+   * Used to position the active indicator for animations of the active indicator on active tab changes.
+   */
+  private activeTabOffset = 0;
+  /**
+   * activeTabScale
+   * Used to scale the tab active indicator up or down as animations of the active indicator occur.
+   */
+  private activeTabScale = 1;
+
+  /**
+   * styles
+   * used in the class for storing the css variables required for animations
+   */
+  private styles: ElementStyles | undefined;
+
+  /**
+   * appearance
+   * There are two modes of appearance: transparent and subtle.
+   */
+  @attr
+  public appearance?: TabsAppearance = TabsAppearance.transparent;
+
+  /**
+   * disabled
+   * Used for disabling all click and keyboard events for the tabs, child tab elements and tab panel elements. UI styling of content and tabs will appear as "grayed out."
+   */
+  @attr({ mode: 'boolean' })
+  public disabled?: boolean;
+
+  /**
+   * size
+   * defaults to medium.
+   * Used to set the size of all the tab controls, which effects text size and margins. Three sizes: small, medium and large.
+   */
+  @attr
+  public size?: TabsSize;
+
+  /**
+   * calculateAnimationProperties
+   *
+   * Recalculates the active tab offset and scale.
+   * These values will be applied to css variables that control the tab active indicator position animations
+   */
+  private calculateAnimationProperties(tab: Tab) {
+    this.activeTabOffset = this.getTabPosition(tab);
+    this.activeTabScale = this.getTabScale(tab);
+  }
+
+  /**
+   * getSelectedTabPosition - gets the x or y coordinates of the tab
+   */
+  private getTabPosition(tab: Tab): number {
+    if (this.orientation === TabsOrientation.horizontal) {
+      return this.previousActiveTabData.x - (tab.getBoundingClientRect().x - this.getBoundingClientRect().x);
+    } else return this.previousActiveTabData.y - (tab.getBoundingClientRect().y - this.getBoundingClientRect().y);
+  }
+
+  /**
+   * getSelectedTabScale - gets the scale of the tab
+   */
+  private getTabScale(tab: Tab): number {
+    if (this.orientation === TabsOrientation.horizontal) {
+      return this.previousActiveTabData.width / tab.getBoundingClientRect().width;
+    } else return this.previousActiveTabData.height / tab.getBoundingClientRect().height;
+  }
+
+  /**
+   * applyUpdatedCSSValues
+   *
+   * calculates and applies updated values to CSS variables
+   * @param tab
+   */
+  private applyUpdatedCSSValues(tab: Tab) {
+    this.calculateAnimationProperties(tab);
+    this.setTabScaleCSSVar();
+    this.setTabOffsetCSSVar();
+  }
+
+  /**
+   * animationLoop
+   * runs through all the operations required for setting the tab active indicator to its starting location, ending location, and applying the animated css class to the tab.
+   * @param tab
+   */
+  private animationLoop(tab: Tab) {
+    // remove the animated class so nothing animates yet
+    tab.classList.remove('animated');
```

Comment:
Not to mention that if someone has a class on the tab the tab would also need to use `:classList` otherwise I'm not sure they would concatenate.

Reply:
Responded offline but I don't see why this needs to be observable, so no need for the attr. This is a hook you're applying to manage styling/animation instead of a class.

Reply:
Thats why data-attributes exist, its really just used for styling when these components are used together. Now if animate was an attribute that was meant to be controlled by the consumer ( something they could toggle on and off) depending on use case then an attribute would make sense. Even then the attribute would only be on the parent `tabs` component and then just assign `data-animate` to the individual `tab`

`HTML is designed with extensibility in mind for data that should be associated with a particular element but need not have any defined meaning. [data-* attributes](https://developer.mozilla.org/en-US/docs/Web/HTML/Global_attributes/data-*) allow us to store extra information on standard, semantic HTML elements without other hacks such as non-standard attributes, or extra properties on DOM.`

Reply:
https://developer.mozilla.org/en-US/docs/Learn/HTML/Howto/Use_data_attributes

Reply:
> The reason I questioned it was from a readability perspective. Should I make it clear at the top of the class that there is a data attribute that's part of the component's inner workings. Then I thought, maybe that's just overkill.

Yeah, I see these as internals just as I see the class usage as internals which didn't need to be documented.